demo_numpy.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import dimod
+from dwave.system import LeapHybridSampler
+
+import demo
+
+def build_bqm(potential_new_cs_nodes, num_poi, pois, num_cs, charging_stations, num_new_cs):
+    """ Build bqm that models our problem scenario using NumPy. """
+
+    # Tunable parameters
+    gamma1 = len(potential_new_cs_nodes) * 4.
+    gamma2 = len(potential_new_cs_nodes) / 3.
+    gamma3 = len(potential_new_cs_nodes) * 1.7
+    gamma4 = len(potential_new_cs_nodes) ** 3
+
+    # Build BQM using adjVectors to find best new charging location s.t. min
+    # distance to POIs and max distance to existing charging locations
+    linear = np.zeros(len(potential_new_cs_nodes))
+
+    pncs = np.asarray(potential_new_cs_nodes)
+    pnp = np.asarray(pois)
+    csnp = np.asarray(charging_stations)
```

Comment:
Up to you, but I find these shorter variables names to be potentially confusing. Perhaps something like `nodes_array`,`pois_array`, and `cs_array`? Though these are much longer.